sentry/src/test/java/io/sentry/hints/HintsTest.kt

```diff
@@ -0,0 +1,101 @@
+package io.sentry.hints
+
+import io.sentry.hints.AttachmentsTest.Companion.newAttachment
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+
+class HintsTest {
+    @Test
+    fun `getting as wrong class returns null`() {
+        val hints = Hints()
+        hints.set("hint1", "not a number")
+
+        assertNull(hints.getAs("hint1", Int::class.java))
+    }
+
+    @Test
+    fun `getting as correct class returns it`() {
+        val hints = Hints()
+        hints.set("hint1", "some string")
+
+        assertEquals("some string", hints.getAs("hint1", String::class.java))
+    }
+
+    @Test
+    fun `getting casted returns null if not contained`() {
+        val hints = Hints()
+        assertNull(hints.getAs("hint-does-not-exist", Int::class.java))
+    }
+
+    @Test
+    fun `getting returns null if not contained`() {
+        val hints = Hints()
+        assertNull(hints.get("hint-does-not-exist"))
+    }
+
+    @Test
+    fun `kotlin java interop for primitives does not work yet`() {
+        val hints = Hints()
+
+        hints.set("hint1", 1718)
```

Comment:
I think `1718` is an integer here, what if you make it `1718L` ?

Reply:
Oops forgot to change it back to Long after playing around. Sadly this makes no difference.

Reply:
Did something similar, code is now updated.